rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,535 @@
+                event.counterparty = CPT_SAI
+                return True
+
+            if event.asset == self.peth:
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Increase CDP collateral by {event.balance.amount} PETH'
+                event.counterparty = CPT_SAI
+                return True
+
+        if (
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            event.counterparty == MAKERDAO_SAITUB_CONTRACT
+        ):
+            if event.asset == self.weth:
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Withdraw {event.balance.amount} WETH from SAI vault'
+                event.counterparty = CPT_SAI
+                return True
+
+            if event.asset == self.peth:
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Decrease CDP collateral by {event.balance.amount} PETH'
+                event.counterparty = CPT_SAI
+                return True
+
+        return False
+
+    def _decode_peth_mint_after_deposit(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],  # pylint: disable=unused-argument
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        """
+        This method decodes the receipt of PETH after WETH has been deposited.
+
+        An example of such transaction is:
+        https://etherscan.io/tx/0x5a7849ab4b7f7de2b005deddef24a094387c248c3bcb06066109bd7852c1d8af
+        """
+        if tx_log.topics[0] == PETH_MINT_EVENT_TOPIC:
+            owner = hex_or_bytes_to_address(tx_log.topics[1])
+            amount_raw = hex_or_bytes_to_int(tx_log.data[:32])
+            amount = asset_normalized_value(amount=amount_raw, asset=self.peth)
+
+            event = HistoryBaseEntry(
+                event_identifier=transaction.tx_hash,
+                sequence_index=self.base_tools.get_sequence_index(tx_log),
+                timestamp=ts_sec_to_ms(transaction.timestamp),
+                location=Location.BLOCKCHAIN,
+                asset=self.peth,
+                event_type=HistoryEventType.RECEIVE,
+                event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+                balance=Balance(amount=amount),
+                location_label=owner,
+                counterparty=CPT_SAI,
+                notes=f'Receive {amount} PETH from Sai Vault',
```

Comment:
Please be consistent. In some events you write it as Sai vault and in some as SAI vault. Pick one and stick to it. I would go with lowercase, as uppercase is the symbol. Please make sure to change all locations.